src/Features/Core/Portable/EditAndContinue/EditSession.cs

```diff
@@ -907,6 +926,15 @@ public async ValueTask<SolutionUpdate> EmitSolutionUpdateAsync(Solution solution
                     }
                 }
 
+                if (hasOutOfSyncChangedDocument)
+                {
+                    // The project is considered stale as long as it has at least one document that is out-of-sync.
+                    // Treat the project the same as if it hasn't been built. We won't produce delta for it until it gets rebuilt.
+                    Log.Write($"Changes not applied to {newProject.Name} '{newProject.FilePath}': binaries not up-to-date");
```

Comment:
fwiw, "binaries not up to date" didn't seem particularly clear to me.  but htis is your domain, so if this makes sense, go for it.